hooks/useFetch.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from "react";
+
+export function useFetch(url, defaultOfState, dependency) {
```

Comment:
do you need the defaultOfState variable as a prop or could you just keep it here? I am not sure since I dont know the whole code.